matlab/mars2wgs.m

```diff
@@ -0,0 +1,23 @@
+function [lat, long ] = mars2wgs(marsLat, marsLong)
+% assumption is that inital mars lat longs are close to actual
+% then solves for the local delta and estimates the original lat longs
+% then uses the estimated lat longs to try again
+% this method seems very stable
+% initialize estimates of lat and long
+
+lat = marsLat;
+long = marsLong;
+threshold = 0.0000001;
+
+for i=1:30
+    [tempLat,tempLong] = wgs2gcj(lat, long);
+    deltaLat = lat - tempLat;
+    deltaLong = long - tempLong;
+    lat = marsLat + deltaLat;
+    long = marsLong + deltaLong;
+    if all(abs(tempLat-marsLat)<threshold & abs(tempLong-marsLong)<threshold)
```

Comment:
just compare on delta lat/lon. also high five for rediscovering this. https://en.wikipedia.org/wiki/Restrictions_on_geographic_data_in_China#Reverse_transformation

Reply:
In my implementation, the delta is the difference between mars and real lat longs.. so I have to check the difference between input mars and my test mars to see if that meets the threshold.

Reply:
oh right… instead of doing some iterative improvement for the wgs coords to a fixed-ish point, you are thinking of doing a more precise gcj/wgs difference. my apologies.